src/Compilers/Core/Portable/SourceGeneration/IncrementalGeneratorSyntaxWalker.cs

```diff
@@ -7,31 +7,33 @@
 using System.Collections.Immutable;
 using System.Diagnostics;
 using System.Text;
+using System.Threading;
 using Microsoft.CodeAnalysis.PooledObjects;
 
 namespace Microsoft.CodeAnalysis
 {
     internal sealed class IncrementalGeneratorSyntaxWalker : SyntaxWalker
     {
-        private readonly Func<SyntaxNode, bool> _filter;
-
+        private readonly Func<SyntaxNode, CancellationToken, bool> _filter;
+        private readonly CancellationToken _token;
         private ArrayBuilder<SyntaxNode>? _results;
 
-        internal IncrementalGeneratorSyntaxWalker(Func<SyntaxNode, bool> filter)
+        internal IncrementalGeneratorSyntaxWalker(Func<SyntaxNode, CancellationToken, bool> filter, CancellationToken token)
         {
             _filter = filter;
+            _token = token;
         }
 
-        public static ImmutableArray<SyntaxNode> GetFilteredNodes(SyntaxNode root, Func<SyntaxNode, bool> func)
+        public static ImmutableArray<SyntaxNode> GetFilteredNodes(SyntaxNode root, Func<SyntaxNode, CancellationToken, bool> func, CancellationToken token)
         {
-            var walker = new IncrementalGeneratorSyntaxWalker(func);
+            var walker = new IncrementalGeneratorSyntaxWalker(func, token);
             walker.Visit(root);
             return walker._results.ToImmutableOrEmptyAndFree();
         }
 
         public override void Visit(SyntaxNode node)
         {
```

Comment:
Is there a reason we don't do a `_token.ThrowIfCancellationRequested()` call here? As written seems like we completely rely on the generator to be cancellation correct. Seems like it would be more robust if we helped out here.

Reply:
We do check cancellation before we get here as we pass the token into the GetRoot() method. That doesn't check between inputs though. I'll add the check in the `DriverStateTable.Builder` so it gets handled in both the current syntax nodes (and any thing else we add) rather than checking in multiple places. 

Tbf we should probably go through and audit where we check the token (the transforms come to mind), but i'll tack that on to a more general issue we have around respecting cancellation https://github.com/dotnet/roslyn/issues/46528